src/RSAccessor/RSPortalAccessorV2.cs

```diff
@@ -101,6 +101,41 @@ public void AddToCatalogItems(CatalogItem item)
             return item.Path;
         }
 
+        public void UpdateDataSourceCredentials(
+            string path,
+            string username,
+            string password,
+            bool isWindowsCredentials)
+        {
+            var ctx = CreateContext();
+            var report = ctx.CatalogItemByPath(path).Expand("DataSources").GetValue();
+
+            var powerBiReport = report as PowerBIReport;
+            if (powerBiReport != null)
+            {
+                foreach (var dataSource in powerBiReport.DataSources)
+                {
+                    dataSource.CredentialRetrieval = CredentialRetrievalType.Store;
+
+                    var credentials = dataSource.CredentialsInServer;
+                    if (credentials == null)
+                    {
+                        dataSource.CredentialsInServer = credentials = new CredentialsStoredInServer();
```

Comment:
That's what I initially thought. But that isn't the case when a property is initially null.